sources/packages/api/src/services/application/application.service.ts

```diff
@@ -75,6 +82,10 @@ export class ApplicationService extends RecordDataModelService<Application> {
     this.logger.log("[Created]");
   }
 
+  setBypassApplicationSubmitValidations(payload: boolean): void {
```

Comment:
do we need this? we can directly use `this.config.bypassApplicationSubmitValidations` right? or am i missing anything

Reply:
I used that before but I had code smells caused by more than 7 parameters in the validateOverlappingDatesAndPIR method. I also tried injecting ConfigService in constructor and still had sonar code smell in because we already have more than 7 injections in some of the constructors of the controller where it is required.

Reply:
@henryoforeh-aot  If the code smell is only because of the number of injected parameters we can override Sonarcloud validations before doing the merge. If not necessary, we can avoid doing changes just to make the Sonarcloud pass 😉

Reply:
maybe to resolve the code smell you can pass student details as a single object, something like `async validateOverlappingDatesAndPIR(
    applicationId: number,
    {
         lastName: string,
         userId: number,
         sin: string,
         birthDate: Date,
    },
         studyStartDate: Date,
         studyEndDate: Date,
  )`

Reply:
@ann-aot my understanding is that the Sonarcloud is complaining about the constructor, not the method. Am I missing something?

Reply:
You're right. Sonar is complaining only about the constructor.